src/js/components/Header/AccountMenu/ModalContents/ConnectWalletContent.tsx

```diff
@@ -0,0 +1,158 @@
+import { ArrowBackIcon, CheckCircleIcon } from '@chakra-ui/icons';
+import {
+    Flex,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent, 
+    ModalFooter, 
+    ModalHeader,
+    Spacer,
+    Text,
+    Button,
+    Image,
+    IconButton,
+    Box,
+    Spinner,
+    Center
+} from '@chakra-ui/react';
+import * as React from 'react';
+import useGlobalSettings from '../../../../GlobalSettings/useGlobalSettings';
+import { IConnectWalletContentProps } from '../../../../types/Header/ContentProps/ContentProps';
+import useAccount from '../../../AccountProvider/useAccount';
+import useMetamask from '../../../ConnectMetamask/useMetamask';
+import useEthereumProvider from '../../../EthereumProvider/useEthereumProvider';
+import AlertDialogMetamask from '../../../Wallet/AlertDialogMetamask/AlertDialogMetamask';
+
+
+
+function ConnectWalletContent({setIsConnectModalOpen, setIsAccountModalOpen}: IConnectWalletContentProps): JSX.Element {
+    
+    const {hardware, isPhoneHardware, connectWay} = useGlobalSettings();
+    const {providerState} = useEthereumProvider();
+    const {connectMetamask} = useMetamask();
+    const {account} = useAccount();
+    const [isConnecting, setIsConnecting] = React.useState<boolean>(false);
+    const [isAlertDialogOpen, setIsAlertDialogOpen] = React.useState<boolean>(false);
+
+
+    const onCloseAlertDialog = React.useCallback(() => {
+
+        setIsAlertDialogOpen(false)
+        setIsConnecting(false);
+    }, []);
+
+    const onMetamask = async () => {
+
+        if(providerState) return;
+
+        setIsConnecting(previous => previous = true);
+
+        await connectMetamask((isProviderDetected: boolean) => {
+            
+            if(isProviderDetected === false) {
+                
+                setIsAlertDialogOpen(true);
+                return;
+            }
+        });
+    }
+
+    const turnOffConnecting = () => {
+
+        if(isConnecting === false) return;
+
+        setIsConnecting(previous => previous = false);
+        setIsAccountModalOpen(previous => previous = true);
+        setIsConnectModalOpen(previous => previous = false);
```

Comment:
What is it? Should it be comparing like `...(previous => previous === false)`? 
Or is it just `setIs.....(false)`?